src/sentry/integrations/vercel/integration.py

```diff
@@ -248,55 +248,60 @@ def update_organization_config(self, data):
+                dsn_env_name: {"type": "plain", "value": sentry_project_dsn},
+                "SENTRY_AUTH_TOKEN": {"type": "secret", "value": sentry_auth_token},
+                "VERCEL_GIT_COMMIT_SHA": {"type": "system", "value": "VERCEL_GIT_COMMIT_SHA"},
+            }
 
-            secrets.append("")
-            for secret, env_var in zip(secrets, env_var_names):
-                self.create_env_var(vercel_client, vercel_project_id, env_var, secret)
+            for env_var, details in env_var_map.items():
+                if details["type"] == "secret":
+                    secret_name = env_var + "_%s" % uuid
+                    secret_value = vercel_client.create_secret(secret_name, details["value"])
+                    self.create_env_var(
+                        vercel_client, vercel_project_id, env_var, secret_value, details["type"]
+                    )
+                else:
+                    self.create_env_var(
+                        vercel_client, vercel_project_id, env_var, details["value"], details["type"]
+                    )
 
         config.update(data)
         self.org_integration.update(config=config)
 
-    def env_var_already_exists(self, client, vercel_project_id, name):
-        return any(
-            [
-                env_var
-                for env_var in client.get_env_vars(vercel_project_id)
-                if env_var["key"] == name
-            ]
-        )
+    def create_env_var(self, client, vercel_project_id, key, value, type):
+        data = {
+            "key": key,
+            "value": value,
+            "target": ["production"],
+            "type": type,
+        }
+        try:
+            return client.create_env_variable(vercel_project_id, data)
+        except ApiError as e:
+            if e.json and e.json.get("error", {}).get("code") == "ENV_ALREADY_EXISTS":
+                return self.update_env_variable(client, vercel_project_id, data)
+            raise
+
+    def update_env_variable(self, client, vercel_project_id, data):
+        key = data["key"]
+        try:
+            envs = client.get_env_vars(vercel_project_id)["envs"]
+        except ApiError:
+            message = f"Failed to get ID for environment variable {key} in Vercel project {vercel_project_id}."
+            raise IntegrationError(message)
 
-    def create_env_var(self, client, vercel_project_id, key, value):
-        if not self.env_var_already_exists(client, vercel_project_id, key):
-            return client.create_env_variable(vercel_project_id, key, value)
-        client.update_env_variable(vercel_project_id, key, value)
+        env_var_ids = [env_var["id"] for env_var in envs if env_var["key"] == data["key"]]
+        if env_var_ids:
+            try:
+                return client.update_env_variable(vercel_project_id, env_var_ids[0], data)
+            except ApiError:
+                message = f"Could not update environment variable {key} in Vercel project {vercel_project_id}."
```

Comment:
@ceorourke My bad for not seeing this before. If we catch the error here and don't actually send the original error to Sentry, we'll never know what actually failed. That's going to make debugging much harder. I'm wondering, why are we catching an `ApiError` here and raising  `IntegrationError`?  Is it so we can display this error to the user?